------------------------------------------------------------------------
r138 | tomo | 2007-09-20 19:26:20 -0400 (Thu, 20 Sep 2007) | 12 lines

From: Arne Redlich <agr@powerkom-dd.de>
Subject: 2.6.23 compile fix
Date: Thu, 20 Sep 2007 23:30:19 +0200

kmem_cache_create()'s arguments will change with 2.6.23.

Note that instead of simply modifying the kmem_cache_create() calls to
match the new interface, they're replaced with KMEM_CACHE(), which was
introduced with 2.6.22 to simplify slab cache creation.

Merely compile tested.

------------------------------------------------------------------------
Index: kernel/iscsi.c
===================================================================
--- kernel/iscsi.c	(revision 137)
+++ kernel/iscsi.c	(revision 138)
@@ -1757,8 +1757,7 @@ static int iscsi_init(void)
 	if ((err = event_init()) < 0)
 		goto err;
 
-	iscsi_cmnd_cache = kmem_cache_create("iscsi_cmnd", sizeof(struct iscsi_cmnd),
-					     0, 0, NULL, NULL);
+	iscsi_cmnd_cache = KMEM_CACHE(iscsi_cmnd, 0);
 	if (!iscsi_cmnd_cache)
 		goto err;
 
Index: kernel/tio.c
===================================================================
--- kernel/tio.c	(revision 137)
+++ kernel/tio.c	(revision 138)
@@ -110,8 +110,7 @@ int tio_sync(struct iet_volume *lu, stru
 
 int tio_init(void)
 {
-	tio_cache = kmem_cache_create("tio", sizeof(struct tio),
-				      0, 0, NULL, NULL);
+	tio_cache = KMEM_CACHE(tio, 0);
 	return  tio_cache ? 0 : -ENOMEM;
 }
 
